test/hotspot/jtreg/compiler/lib/template_framework/StructuralName.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.lib.template_framework;
+
+import java.util.List;
+
+/**
+ * {@link StructuralName}s represent things like method and class names, and can be added to the local
+ * scope with {@link Template#addStructuralName}, and accessed with {@link Template#structuralNames}, from where
+ * count, list or even sample random {@link StructuralName}s. Every {@link StructuralName} has a {@link StructuralName.Type},
+ * so that sampling can be restricted to these types.
+ *
+ * For field and variable names and alike, there are the analogous {@link DataName}s.
+ *
+ * @param name The {@link String} name used in code.
+ * @param type The type of the {@link StructuralName}.
+ * @param weight The weight of the {@link StructuralName}, it corresponds to the probability of choosing this
+ *               {@link StructuralName} when sampling later on.
+ */
+public record StructuralName(String name, StructuralName.Type type, int weight) implements Name {
+
+    /**
+     * Creates a new {@link StructuralName}.
+     */
+    public StructuralName {
+    }
```

Comment:
Is this required? Is it not automatically added? Same for `DataName`.

Reply:
If I remove it, then `javadoc` complains that we are now using the default constructor, and that it does not have a comment for it... kinda strange but ok :sweat_smile:

Reply:
So I'd rather keep it. It's a bit of unnecessary boilarplate, but I like having `javadoc` all happy and quiet.